recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.57.0"
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    package_type = "static-library"
```

Comment:
Is correct to affirm that pcap++ only supports static library? I see shared option option available in the CMake file: https://github.com/seladb/PcapPlusPlus/blob/v23.09/CMakeLists.txt#L95


Plus, I can build in my machine shared library without errors:

```
$ cmake -S . -B build -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/tmp/install -DCMAKE_VERBOSE_MAKEFILE=ON -DPCAPPP_BUILD_TESTS=OFF -DPCAPPP_BUILD_EXAMPLES=OFF -DCMAKE_TOOLCHAIN_FILE=build/Release/generators/conan_toolchain.cmake
...
[ 27%] Linking CXX shared library libCommon++.so
cd /home/conan/project/build/Common++ && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/Common++.dir/link.txt --verbose=1
/usr/bin/g++ -fPIC -m64 -O3 -DNDEBUG -m64 -shared -Wl,-soname,libCommon++.so.23.09 -o libCommon++.so.23.09 CMakeFiles/Common++.dir/src/GeneralUtils.cpp.o CMakeFiles/Common++.dir/src/IpAddress.cpp.o CMakeFiles/Common++.dir/src/IpUtils.cpp.o CMakeFiles/Common++.dir/src/Logger.cpp.o CMakeFiles/Common++.dir/src/MacAddress.cpp.o CMakeFiles/Common++.dir/src/OUILookup.cpp.o CMakeFiles/Common++.dir/src/PcapPlusPlusVersion.cpp.o CMakeFiles/Common++.dir/src/SystemUtils.cpp.o CMakeFiles/Common++.dir/src/TablePrinter.cpp.o 
cd /home/conan/project/build/Common++ && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_symlink_library libCommon++.so.23.09 libCommon++.so.23.09 libCommon++.so
```

Reply:
Shared libraries are not well supported on all platforms, so I prefer to keep the static option only